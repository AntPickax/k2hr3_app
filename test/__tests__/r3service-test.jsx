/*
 *
 * K2HR3 Web Application
 *
 * Copyright 2017 Yahoo! Japan Corporation.
 *
 * K2HR3 is K2hdkc based Resource and Roles and policy Rules, gathers
 * common management information for the cloud.
 * K2HR3 can dynamically manage information as "who", "what", "operate".
 * These are stored as roles, resources, policies in K2hdkc, and the
 * client system can dynamically read and modify these information.
 *
 * For the full copyright and license information, please view
 * the license file that was distributed with this source code.
 *
 * AUTHOR:   Takeshi Nakatani
 * CREATE:   Mon Jul 10 2018
 * REVISION:
 *
 */

import React					from 'react';							// eslint-disable-line no-unused-vars
import getMuiTheme				from 'material-ui/styles/getMuiTheme';
import renderer					from 'react-test-renderer';
import getElementWithContext	from 'react-test-context-provider';		// for context provider

import R3Service				from '../../src/components/r3service';
import R3Provider				from '../../src/util/r3provider';
import r3Theme					from '../../src/components/r3theme';

import mock_fetch				from '../__mocks__/fetchMock';			// eslint-disable-line no-unused-vars
import { createNodeMock }		from '../__mocks__/materialUiMock';		// for material-ui
import mock_injecttap			from '../__mocks__/injectTapMock';		// eslint-disable-line no-unused-vars

//
// Mock functions
//
// [NOTE]
// If you need to customize return value, you can call methods
// for return value by each mock function.
// see: https://jestjs.io/docs/ja/mock-functions
//
// ex)	mockfunc
//			.mockReturnValueOnce(10)
//			.mockReturnValueOnce('x')
//			.mockReturnValue(true);
//
const save		= jest.fn();											// eslint-disable-line no-undef
const update	= jest.fn();											// eslint-disable-line no-undef

//
// Mock functions for parts in TextField
//
// [NOTE]
// Material-ui's TextField has some components these set random "id" to itself.
// The random id is created by Math functions.
//		ex: "Math.floor(Math.random() * 0xFFFF)"
// So we need fixed id to compare snapshots, set following mock function.
//
Math.random = jest.fn(() => 0);											// eslint-disable-line no-undef

//
// Dummy datas
//
const service = {
	name:		'JEST_SERVICE_NAME',
	owner:		'9999',
	tenant:		[ 'yrn:yahoo:::28862', 'yrn:yahoo:::37146' ],
	verify:		'http://localhost:3000/v1/test/verify'
};

//
// Main test
//
describe('R3Service', () => {											// eslint-disable-line no-undef
	it('test snapshot for R3Service', () => {							// eslint-disable-line no-undef
		/* eslint-disable indent */
		const r3provider	= new R3Provider(null);
		const element		= getElementWithContext({
									muiTheme:	getMuiTheme(r3Theme),
									r3Context:	r3provider.getR3Context()
								},
								<R3Service
									tenant={ 'JEST_TENANT_NAME' }
									service={ service }
									r3provider={ r3provider }
									onSave={ save }
									onUpdate={ update }
								/>
							);
		/* eslint-enable indent */

		const component = renderer.create(element, { createNodeMock });
		let tree		= component.toJSON();
		expect(tree).toMatchSnapshot();									// eslint-disable-line no-undef
	});
});

/*
 * VIM modelines
 *
 * vim:set ts=4 fenc=utf-8:
 *
 */
