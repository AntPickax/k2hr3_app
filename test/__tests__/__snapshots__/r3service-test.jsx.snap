// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`R3Service test snapshot for R3Service 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div>
    <div
      style={
        Object {
          "marginTop": "48px",
        }
      }
    >
      <span
        style={
          Object {
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "14px",
            "fontWeight": "bold",
          }
        }
      >
        VERIFY URL or STATIC JSON OBJECT STRING
      </span>
    </div>
    <div
      style={
        Object {
          "marginLeft": "64px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 48,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 0,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Input Text string or false or Object formatted by JSON
        </div>
        <input
          disabled={false}
          id="verifyvalue-InputTextstringorfalseorObjectformattedbyJSON-undefined-0"
          name="verify_value"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "color": "#212121",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
          value="http://localhost:3000/v1/test/verify"
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#9575cd",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#673ab7",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "marginTop": "48px",
        }
      }
    >
      <span
        style={
          Object {
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "14px",
            "fontWeight": "bold",
          }
        }
      >
        TENANTS
      </span>
    </div>
    <div
      style={
        Object {
          "marginLeft": "64px",
        }
      }
    >
      <div
        className="clearof"
      >
        <div
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "14px",
              "height": 48,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "calc(100% - 60px)",
            }
          }
        >
          <input
            disabled={false}
            id="tenants0-undefined-undefined-0"
            name="tenants_0"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "color": "#212121",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="text"
            value="yrn:yahoo:::28862"
          />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 1,
                  "borderColor": "#9575cd",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 2,
                  "borderColor": "#673ab7",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "border": 10,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "float": "right",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 0,
              "fontWeight": "inherit",
              "height": 48,
              "margin": 0,
              "marginLeft": 6,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "visible",
              "padding": 0,
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": 48,
              "zIndex": 1,
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "color": "#ffffff",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "10px",
                  "left": -26,
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "opacity": 0,
                  "overflow": "hidden",
                  "padding": "0 8px",
                  "position": "absolute",
                  "right": null,
                  "top": -10000,
                  "transition": "top 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms, transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, opacity 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 3000,
                }
              }
            >
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": "50%",
                    "left": "50%",
                    "muiPrepared": true,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translate(-50%, -50%)",
                    "transition": "width 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms, height 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms, backgroundColor 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              />
              <span
                style={
                  Object {
                    "muiPrepared": true,
                    "position": "relative",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Delete
              </span>
            </div>
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#607d8b",
                  "display": "inline-block",
                  "fontSize": 24,
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                }
              }
            />
          </div>
        </button>
      </div>
      <div
        className="clearof"
      >
        <div
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "14px",
              "height": 48,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "calc(100% - 60px)",
            }
          }
        >
          <input
            disabled={false}
            id="tenants1-undefined-undefined-0"
            name="tenants_1"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "color": "#212121",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="text"
            value="yrn:yahoo:::37146"
          />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 1,
                  "borderColor": "#9575cd",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 2,
                  "borderColor": "#673ab7",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "border": 10,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "float": "right",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 0,
              "fontWeight": "inherit",
              "height": 48,
              "margin": 0,
              "marginLeft": 6,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "visible",
              "padding": 0,
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": 48,
              "zIndex": 1,
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "color": "#ffffff",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "10px",
                  "left": -26,
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "opacity": 0,
                  "overflow": "hidden",
                  "padding": "0 8px",
                  "position": "absolute",
                  "right": null,
                  "top": -10000,
                  "transition": "top 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms, transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, opacity 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 3000,
                }
              }
            >
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": "50%",
                    "left": "50%",
                    "muiPrepared": true,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translate(-50%, -50%)",
                    "transition": "width 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms, height 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms, backgroundColor 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              />
              <span
                style={
                  Object {
                    "muiPrepared": true,
                    "position": "relative",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Delete
              </span>
            </div>
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#607d8b",
                  "display": "inline-block",
                  "fontSize": 24,
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                }
              }
            />
          </div>
        </button>
      </div>
      <div
        className="clearof"
      >
        <div
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "14px",
              "height": 48,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "calc(100% - 60px)",
            }
          }
        >
          <div
            style={
              Object {
                "bottom": 12,
                "color": "rgba(0, 0, 0, 0.3)",
                "muiPrepared": true,
                "opacity": 1,
                "position": "absolute",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            Tenant YRN path
          </div>
          <input
            disabled={false}
            id="tenantnew-TenantYRNpath-undefined-0"
            name="tenant_new"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "color": "#212121",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 1,
                  "borderColor": "#9575cd",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 2,
                  "borderColor": "#673ab7",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "border": 10,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "float": "right",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 0,
              "fontWeight": "inherit",
              "height": 48,
              "margin": 0,
              "marginLeft": 6,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "visible",
              "padding": 0,
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": 48,
              "zIndex": 1,
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "color": "#ffffff",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "10px",
                  "left": -26,
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "opacity": 0,
                  "overflow": "hidden",
                  "padding": "0 8px",
                  "position": "absolute",
                  "right": null,
                  "top": -10000,
                  "transition": "top 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms, transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, opacity 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 3000,
                }
              }
            >
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": "50%",
                    "left": "50%",
                    "muiPrepared": true,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translate(-50%, -50%)",
                    "transition": "width 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms, height 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms, backgroundColor 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              />
              <span
                style={
                  Object {
                    "muiPrepared": true,
                    "position": "relative",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Add
              </span>
            </div>
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#607d8b",
                  "display": "inline-block",
                  "fontSize": 24,
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                }
              }
            />
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "48px",
      }
    }
  >
    <div
      className="clearof"
      style={
        Object {
          "float": "right",
        }
      }
    >
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ede7f6",
            "borderRadius": 2,
            "boxShadow": undefined,
            "boxSizing": "border-box",
            "color": "#212121",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "margin": 4,
            "minWidth": 88,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#d1c4e9",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "default",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
              "zIndex": 1,
            }
          }
          tabIndex={-1}
          type="button"
        >
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <span
              color="#ede7f6"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ede7f6",
                  "display": "inline-block",
                  "fontSize": 24,
                  "marginLeft": 12,
                  "marginRight": 0,
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "verticalAlign": "middle",
                }
              }
            />
            <span
              style={
                Object {
                  "color": "#ede7f6",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "margin": 0,
                  "muiPrepared": true,
                  "opacity": 1,
                  "paddingLeft": 8,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "userSelect": "none",
                }
              }
            >
              CANCEL
            </span>
          </div>
        </button>
      </div>
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ede7f6",
            "borderRadius": 2,
            "boxShadow": undefined,
            "boxSizing": "border-box",
            "color": "#212121",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "margin": 4,
            "minWidth": 88,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#d1c4e9",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "default",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
              "zIndex": 1,
            }
          }
          tabIndex={-1}
          type="button"
        >
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <span
              color="#ede7f6"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ede7f6",
                  "display": "inline-block",
                  "fontSize": 24,
                  "marginLeft": 12,
                  "marginRight": 0,
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "verticalAlign": "middle",
                }
              }
            />
            <span
              style={
                Object {
                  "color": "#ede7f6",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "margin": 0,
                  "muiPrepared": true,
                  "opacity": 1,
                  "paddingLeft": 8,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "userSelect": "none",
                }
              }
            >
              SAVE
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
